app/js/enhancedDate.js

```diff
@@ -1,5 +1,156 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
```

Comment:
You don't need to pass date in here. The wrapping function in the immediately invoked function expression is only there in this case to provide scope for the module. There's nothing being passed in where it's actually invoked (at the bottom of the module) so date will just undefined. Where you want to put a parameter instead is in the setDate method: dateMethods.setDate = function (date) { }